src/shared_modules/indexer_connector/testtool/main.cpp

```diff
@@ -1,40 +1,152 @@
-
-    indexerConnector.publish(R"(
+        result += alphanum[std::rand() % (sizeof(alphanum) - 1)];
+    }
+    return result;
+}
+
+float generateRandomFloat(float min, float max)
+{
+    return min + static_cast<float>(std::rand()) / (static_cast<float>(RAND_MAX / (max - min)));
+}
+
+int generateRandomInt(int min, int max)
+{
+    return min + (rand() % static_cast<int>(max - min + 1));
+}
+
+// Generate timestamp.
+std::string generateTimestamp()
+{
+    std::time_t t = std::time(nullptr);
+    std::tm tm = *std::localtime(&t);
+    std::stringstream ss;
+    ss << std::put_time(&tm, "%Y-%m-%dT%H:%M:%S");
+    return ss.str();
+}
+
+nlohmann::json fillWithRandomData(const nlohmann::json& templateJson)
+{
+    nlohmann::json result;
+
+    for (auto& [key, value] : templateJson.items())
     {
-        "type": "test-index",
-        "data": {
-            "message": "Hello world!"
-        },
-        "id": "1",
-        "operation": ""
-    })");
-
-    indexerConnector.publish(R"(
+        if (value.is_object())
+        {
+            if (key == "properties")
+            {
+                result.merge_patch(fillWithRandomData(value));
+            }
+            else
+            {
+                result[key] = fillWithRandomData(value);
+            }
+        }
+        else if (key == "type")
+        {
+            if (value.get<std::string>() == "keyword")
+            {
+                result = generateRandomString(10);
+            }
+            else if (value.get<std::string>() == "long")
+            {
+                result = generateRandomInt(0, 1000);
+            }
+            else if (value.get<std::string>() == "float")
+            {
+                result = generateRandomFloat(0.0, 100.0);
+            }
+            else if (value.get<std::string>() == "date")
+            {
+                result = generateTimestamp();
+            }
+        }
+    }
+
+    return result;
+}
+
+int main(const int argc, const char* argv[])
+{
+    try
+    {
+        CmdLineArgs cmdArgParser(argc, argv);
+
+        // Read configuration file.
+        std::ifstream configurationFile(cmdArgParser.getConfigurationFilePath());
+        if (!configurationFile.is_open())
+        {
+            throw std::runtime_error("Could not open configuration file.");
+        }
+        nlohmann::json configuration;
+        configurationFile >> configuration;
+
+        // Create indexer connector.
+        IndexerConnector indexerConnector(configuration, cmdArgParser.getTemplateFilePath());
+        // Read events file.
+        // If the events file path is empty, then the events are generated
+        // automatically.
+        if (!cmdArgParser.getEventsFilePath().empty())
+        {
+            std::ifstream eventsFile(cmdArgParser.getEventsFilePath());
+            if (!eventsFile.is_open())
+            {
+                throw std::runtime_error("Could not open events file.");
+            }
+            nlohmann::json events;
```

Comment:
Coding style: the `nlohmann` variable can be initialized directly from the `std::ifstream` object.